runtime/drivers/admin/admin.go

```diff
@@ -276,7 +278,10 @@ func (h *Handle) checkHandshake(ctx context.Context) error {
 	}
 
 	if h.repoPath == "" {
-		h.repoPath, err = os.MkdirTemp("", "admin_driver_repo")
+		if err := os.Mkdir(h.config.DataDir, os.ModePerm); err != nil && !errors.Is(err, fs.ErrExist) {
+			return err
+		}
+		h.repoPath, err = os.MkdirTemp(h.config.DataDir, "admin_driver_repo")
```

Comment:
It's problematic to create temporary files in the `data_dir/instance_id/connector` directory because it makes it hard to clear out old `tmp` files.

Probably we should centralize temp files in `data_dir/instance/tmp/` or maybe even `data_dir/tmp/`? And clear the `tmp` directory each time the runtime is started (since when not using the system temp folder, it doesn't get auto-cleared).

Regarding accessing it here in the connector, we could pass in both a `DataDir` and a `TempDir`?

Reply:
I thought we should rely on connectors to delete any temporary files which they should be able to do in most cases apart from crashes.
Storing temporary files in a separate directory is also a good idea but I would prefer to keep `data_dir/instance/tmp` for following reasons even if it will add extra code to delete temporary files.

1. Max storage space will not track downloaded data in case we want to track that. 
2. Also I believe the main purpose of this exercise was to make sure that we add all data for an instance in one directory only.

Reply:
> I thought we should rely on connectors to delete any temporary files

On the happy path, yes, but things crash and have bugs. We should prevent temp data from accumulating in a persistent directory over time.

> I would prefer to keep `data_dir/instance/tmp`

Yeah you're right, of course we need to use that to ensure correct disk usage metrics :)